File: server/java/src/main/java/com/server/reveal/UserContextProvider.java
package com.server.reveal;

import com.infragistics.reveal.sdk.api.IRVUserContext;
import com.infragistics.reveal.sdk.base.RVUserContext;
import com.infragistics.reveal.sdk.rest.RVContainerRequestAwareUserContextProvider;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import java.util.HashMap;
import jakarta.ws.rs.container.ContainerRequestContext;

@Component
public class UserContextProvider extends RVContainerRequestAwareUserContextProvider {
    
    @Value("${REDSHIFT_HOST}")
    private String redshiftHost;
    
    @Value("${REDSHIFT_DATABASE}")
    private String redshiftDatabase;
    
    @Value("${REDSHIFT_USERNAME}")
    private String redshiftUsername;
    
    @Value("${REDSHIFT_PASSWORD}")
    private String redshiftPassword;
    
    @Value("${REDSHIFT_SCHEMA}")
    private String redshiftSchema;

    @Override
    protected IRVUserContext getUserContext(ContainerRequestContext requestContext) {
        String headerValue = requestContext.getHeaderString("x-header-one");
        String userId = null;
        String orderId = null;

        if (headerValue != null && !headerValue.isEmpty()) {
            String[] pairs = headerValue.split(",");
            for (String pair : pairs) {
                String[] kv = pair.split(":", 2);
                if (kv.length == 2) {
                    String key = kv[0].trim();
                    String value = kv[1].trim();
                    if (key.equalsIgnoreCase("userId")) {
                        userId = value;
                    } else if (key.equalsIgnoreCase("orderId")) {
                        orderId = value;
                    }
                }
            }
        }

        // default to User role
        String role = "User";

        // null is used here just for demo 
        if ("BLONP".equals(userId) || userId == null) {
            role = "Admin";
        }

        String[] filterTables = role.equals("Admin") 
            ? new String[0] 
            : new String[]{"customers", "orders", "order_details"};

        var props = new HashMap<String, Object>();
        props.put("OrderId", orderId != null ? orderId : "");
        props.put("Role", role);
        props.put("Host", redshiftHost);
        props.put("Database", redshiftDatabase);
        props.put("Username", redshiftUsername);
        props.put("Password", redshiftPassword);
        props.put("Schema", redshiftSchema);
        props.put("FilterTables", filterTables);

        return new RVUserContext(userId, props); 
    }
}

File: server/aspnet/RevealSdk.Server/Reveal/UserContextProvider.cs
using Microsoft.Extensions.Options;
using Reveal.Sdk;
using RevealSdk.Server.Configuration;

namespace RevealSdk.Server.Reveal
{

    public class UserContextProvider : IRVUserContextProvider
    {
        private readonly redshiftOptions _sqlOptions;

        public UserContextProvider(IOptions<redshiftOptions> sqlOptions)
        {
            _sqlOptions = sqlOptions.Value;
        }

        IRVUserContext IRVUserContextProvider.GetUserContext(HttpContext aspnetContext)
        {
            string? headerValue = aspnetContext.Request.Headers["x-header-one"].FirstOrDefault();
            string? userId = null;
            string? orderId = null;

            if (!string.IsNullOrEmpty(headerValue))
            {
                var pairs = headerValue.Split(',');
                foreach (var pair in pairs)
                {
                    var kv = pair.Split(':', 2);
                    if (kv.Length == 2)
                    {
                        var key = kv[0].Trim();
                        var value = kv[1].Trim();
                        if (key.Equals("userId", StringComparison.OrdinalIgnoreCase))
                            userId = value;
                        else if (key.Equals("orderId", StringComparison.OrdinalIgnoreCase))
                            orderId = value;
                    }
                }
            }

            // default to User role
            string role = "User";

            // null is used here just for demo 
            if (userId == "BLONP" || userId == null)
            {
                role = "Admin";
            }

            var filterTables = role == "Admin"
                ? Array.Empty<string>()
                : ["customers", "orders", "order_details"];

            var props = new Dictionary<string, object>() {
                { "OrderId", orderId ?? string.Empty },
                { "Role", role },
                { "REDSHIFT_HOST", _sqlOptions.REDSHIFT_HOST },
                { "REDSHIFT_DATABASE", _sqlOptions.REDSHIFT_DATABASE },
                { "REDSHIFT_USERNAME", _sqlOptions.REDSHIFT_USERNAME },
                { "REDSHIFT_PASSWORD", _sqlOptions.REDSHIFT_PASSWORD },
                { "REDSHIFT_SCHEMA", _sqlOptions.REDSHIFT_SCHEMA },
                { "FilterTables", filterTables }
            };

            return new RVUserContext(userId, props);
        }
    }
}